persist/adapter.go

```diff
@@ -65,3 +66,44 @@ type Adapter interface {
 	// This is part of the Auto-Save feature.
 	RemoveFilteredPolicy(sec string, ptype string, fieldIndex int, fieldValues ...string) error
 }
+
+type ContextAdapter interface {
+	Adapter
+
+	// LoadPolicy loads all policy rules from the storage.
+	LoadPolicyWithContext(ctx context.Context, model model.Model) error
+}
+
+type WrappedAdapter struct {
+	a Adapter
+}
+
+func (w *WrappedAdapter) LoadPolicy(model model.Model) error {
+	return w.a.LoadPolicy(model)
+}
+
+func (w *WrappedAdapter) SavePolicy(model model.Model) error {
+	return w.a.SavePolicy(model)
+}
+
+func (w *WrappedAdapter) AddPolicy(sec string, ptype string, rule []string) error {
+	return w.a.AddPolicy(sec, ptype, rule)
+}
+
+func (w *WrappedAdapter) RemovePolicy(sec string, ptype string, rule []string) error {
+	return w.a.RemovePolicy(sec, ptype, rule)
+}
+
+func (w *WrappedAdapter) RemoveFilteredPolicy(sec string, ptype string, fieldIndex int, fieldValues ...string) error {
+	return w.a.RemoveFilteredPolicy(sec, ptype, fieldIndex, fieldValues...)
+}
+
+func (w *WrappedAdapter) LoadPolicyWithContext(ctx context.Context, model model.Model) error {
+	return w.a.LoadPolicy(model)
```

Comment:
I don't quite understand what the wrappadapter does since the incoming ``ctx`` isn't used